### PR TITLE
fix: bosh CLI not stopping on CTRL+C

### PR DESCRIPTION
**Pull request by b1tamara**

### CODE REVIEW

File: src/pcap/client.go [hunk 1]
```diff
@@ -172,7 +169,8 @@ func (c *Client) ProcessCapture(ctx context.Context, endpointRequest *EndpointRe
 		},
 	}
 
-	c.stream, err = c.Capture(ctx)
+	apiContext := context.Background()
+	c.stream, err = c.Capture(apiContext)
```
Comment (maxmoehl) [inline src/pcap/client.go:173]:
```suggestion
	c.stream, err = c.Capture(context.Background())
```

**Review by maxmoehl — CHANGES_REQUESTED** [review]
You mainly aligned the client side implementation with the API->Agent logic, right?

Looks good overall, is there a way we can add a test case for this to catch changes that would break this logic again?

File: src/pcap/client.go [hunk 2]
```diff
@@ -244,18 +247,20 @@ func (c *Client) ReadCaptureResponse(stream API_CaptureClient, packetWriter *pca
...
 			res, err := stream.Recv()
 			if errors.Is(err, io.EOF) {
 				logger.Info("clean stop, done")
 				cancel(nil)
-				break
+				return
 			}
+
 			code := status.Code(err)
 			if code != codes.OK {
```
Comment (peanball) [inline src/pcap/client.go:260]: Do we need to check for an `OK` status as well?

File: src/pcap/client.go [hunk 3]
```diff
@@ -233,6 +234,8 @@ func (c *Client) StopRequest() {
 	if err != nil {
 		c.log.Error("could not stop")
 	}
+	c.stream.CloseSend()
```
[the remark this answers is not in the document]
Comment (b1tamara) [reply src/pcap/client.go:237]: We have a bidi stream. CloseSend closes the send direction (from client to server) of the stream to tell the server that we won’t send any more data.

File: src/pcap/client.go (hunk 2 above)
Comment (b1tamara) [reply src/pcap/client.go:260]: @peanball Why do you think we need the status OK? We are checking here for on err from stream.Recv but if there are no errors, we only read the packet/msg

**Comment by b1tamara** [comment]
@maxmoehl Yes. Now the implementation of client is very similar to api and agent (in functionality of main _capture_ function and in reading from stream)

**Review by maxmoehl — APPROVED** [review]

**Pull request merged**